x-pack/plugins/apm/common/utils/apm_ml_anomaly_query.ts

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+export function apmMlAnomalyQuery(detectorIndex: 0 | 1 | 2) {
+  return [
+    {
+      bool: {
+        filter: [
+          {
+            terms: {
+              result_type: ['model_plot', 'record'],
+            },
+          },
+          {
+            term: { detector_index: detectorIndex },
```

Comment:
@peteharverson Do we use the detector index to distinguish between latency, throughput and transaction failure rate? 
Are we concerned about storage since we don't use a keyword, eg `type: "latency"` which would be more readable?

Reply:
The model plot and record results _do_ store the individual attributes as well as the detector index (for example see https://github.com/elastic/elasticsearch/blob/63012c8a40742829806b44cb6577b05d9ab25a83/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/job/results/ModelPlot.java#L35-L47) but I guess it is more efficient to search for  a single integer rather than all the corresponding attributes in the apm code. Using an enum as you suggested in https://github.com/elastic/kibana/pull/111957#pullrequestreview-769422638 would help to make it more readable.